lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -367,10 +368,13 @@ class DocumentImpl extends NodeImpl {
     this.write(...arguments, "\n");
   }
 
-  // This is implemented separately for Document (which has a _ids cache) and DocumentFragment (which does not).
-  getElementById(id) {
-    // Return the first element with this ID.
-    return this._ids[id] && this._ids[id].length > 0 ? this._ids[id][0] : null;
+  getElementById(elementId) {
```

Comment:
I am worried about the performance degradation here of removing the cache. Have you done any tests?

Reply:
I changed the implementation to use the cached ids and find the first one that is not enclosed in a shadow tree.